command/bucket_version.go

```diff
@@ -0,0 +1,145 @@
+package command
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/peak/s5cmd/log"
+	"github.com/peak/s5cmd/storage"
+	"github.com/peak/s5cmd/storage/url"
+	"github.com/peak/s5cmd/strutil"
+)
+
+var bucketVersionHelpTemplate = `Name:
+	{{.HelpName}} - {{.Usage}}
+
+Usage:
+	{{.HelpName}} [options] s3://bucketname
+
+Options:
+	{{range .VisibleFlags}}{{.}}
+	{{end}}
+Examples:
+	1. Get bucket versioning status of a bucket
+		 > s5cmd {{.HelpName}} s3://bucketname
+
+	2. Enable bucket versioning for the bucket
+		 > s5cmd {{.HelpName}} --set Enabled s3://bucketname
+
+	3. Suspend bucket versioning for the bucket
+		 > s5cmd {{.HelpName}} --set Suspended s3://bucketname
+`
+
+func NewBucketVersionCommand() *cli.Command {
+	return &cli.Command{
+		Name:               "bucket-version",
+		CustomHelpTemplate: bucketVersionHelpTemplate,
+		HelpName:           "bucket-version",
+		Usage:              "configure bucket versioning",
+		Flags: []cli.Flag{
+			&cli.GenericFlag{
+				Name: "set",
+				Value: &EnumValue{
+					Enum:    []string{"Suspended", "Enabled"},
```

Comment:
What do you think of making `Suspended` and `Enabled` case-insensitive? It might be more user-friendly.

Reply:
The API requests is case sensitive.
The `aws s3api` is also case sensitive:
The following fails
```
$ aws s3api put-bucket-versioning --bucket mybucketname --versioning-configuration Status=suspended
````
>An error occurred (MalformedXML) when calling the PutBucketVersioning operation: The XML you provided was not well-formed or did not validate against our published schema

while the following succeds.

```
$ aws s3api put-bucket-versioning --bucket mybucketname --versioning-configuration Status=Suspended
```

mc has a different approach uses subcommands:

```
mc version enable myminio/mydata
mc version info myminio/mydata
mc version suspend myminio/mydata
```

Nevertheless, it could be a nice UX improvement, so I'll give it a try.

Reply:
I made the changes and committed.

Reply:
thank you. it is much better

Reply:
My pleasure, thanks!